source/falaise/snemo/datamodels/angle_measurement.h

```diff
@@ -0,0 +1,65 @@
+/// \file falaise/snemo/datamodels/angle_measurement.h
+/* Author(s) :    Steven Calvez <calvez@lal.in2p3.fr>
+ * Creation date: 2015-10-24
+ * Last modified: 2015-10-24
+ *
+ * Description: The class for angle measurement
+ */
+
+#ifndef FALAISE_SNEMO_DATAMODEL_ANGLE_MEASUREMENT_H
+#define FALAISE_SNEMO_DATAMODEL_ANGLE_MEASUREMENT_H 1
+
+// This project:
+#include <falaise/snemo/datamodels/base_topology_measurement.h>
+
+namespace snemo {
+
+namespace datamodel {
+
+/// \brief The angle measurement
+class angle_measurement : public base_topology_measurement {
+ public:
+  /// Constructor
+  angle_measurement();
```

Comment:
Related to the `set/get/grab_angle` methods, it's not clear if the concrete `measurement` classes are just POD or true classes. There's no real encapsulation of the measurement value.

Reply:
I don't get your point...

Reply:
What's different between this and

```c++
class angle_measurement : public base_topology_measurement {
 public:
  double angle;
};
```

Why not do something like:

```c++
class angle_measurement : public base_topology_measurement {
 public:
  explicit angle_measurement(double angle) : angle_(angle){}
  ...

  double get_angle() const {
    return angle_;
  }
 
 private:
  double angle_ ;
};
```
?

Reply:
Ok. You are perfectly right but this kind of (mis)conception of the code is linked to the way the measurements are actually filled by the `ParticleIdentification` module. This might be not a good reason but this is the actual reason. 

The basic idea in the PID module is to provide a common interface for the different measurement calculations : we first pushed a new measurement into the measurement dictionnary and then, dedicated drivers such as angle, energy, TOF drivers with common interface (basically a `process` method taking `particle_track`(s) + the measurement) perform the calculation itself. I think your proposal will be fine if all the data type of the measurement will be the same (`tof_measurement` is slightly more complicated with a `vector` that you don't want to copy). Here, setting the measurement value is hidden within measurement drivers. The counterpart is that the class exposes its data but changing it, is really too much work in PID. I will prefer to keep it even if it is badly designed.